### PR TITLE
fix: Consistently save Credentials, Connector & admin_session values under session attribute

### PR DESCRIPTION
**Pull request by Jad-el-khoury**
## Description

Currently, the CredentialsFilter saves 3 attributes CONNECTOR_ATTRIBUTE, CREDENTIALS_ATTRIBUTE & ADMIN_SESSION_ATTRIBUTE.

CREDENTIALS_ATTRIBUTE & ADMIN_SESSION_ATTRIBUTE are always saved as a Session attribute.
Yet, CONNECTOR_ATTRIBUTE is sometimes saved as a request attribute, or a session attribute. (or maybe ultimatly, it is saved in both session and request, but this is hard to guess from the code).

Consistently save the CONNECTOR_ATTRIBUT under session attribute, like all other attributes.

## Checklist

- [x] This PR adds an entry to the CHANGELOG. _See https://keepachangelog.com/en/1.0.0/ for instructions. Minor edits are exempt._
- [x] This PR was tested on at least one Lyo OSLC server (e.g. manual workflow on [Lyo Sample](https://github.com/OSLC/lyo-samples/actions/workflows/maven-smoke-manual.yml) and [OSLC RefImpl](https://github.com/oslc-op/refimpl/actions/workflows/maven-acceptance-manual.yml)) or adds unit/integration tests.
- [x] This PR does NOT break the API

### CODE REVIEW

**Comment by Jad-el-khoury** [comment]
@berezovskyi @jamsden  Can you please review? I cannot add reviewers to the request.

**Comment by berezovskyi** [comment]
We will likely need to target this for Lyo 6.0 due to the breaking change.

The CHANGELOG will also have to be updated accordingly then.

File: server/oauth-core/src/main/java/org/eclipse/lyo/server/oauth/core/utils/AbstractAdapterCredentialsFilter.java [hunk 1]
```diff
@@ -308,7 +361,9 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
 					} catch (UnauthorizedException e)
 					{
 						sendUnauthorizedResponse(response, e);
-						System.err.println(e.getMessage());
+						//TODO: Change to a log message.
+			            log.info("UnauthorizedException occured while checking for Basic authentication: {}", e.getMessage());
```
Comment (berezovskyi) [inline server/oauth-core/src/main/java/org/eclipse/lyo/server/oauth/core/utils/AbstractAdapterCredentialsFilter.java]: make sure the password is not in the message.

suggest to use DEBUG level.

**Comment by Jad-el-khoury** [comment]
Is this really a breaking change?
Notice that the doFilter method saves both attributes under the session at the end of the method.
The main change I am making is in the public  `getConnector()` method, where I am returning the value from the session instead of the request.  More of a bug fix, than a change?

```
session.setAttribute(CONNECTOR_ATTRIBUTE, connector);
session.setAttribute(CREDENTIALS_ATTRIBUTE, credentials);

```
https://github.com/eclipse/lyo/blob/8d33a43f64f81b80c1a2f8dd0001f2606abba261/server/oauth-core/src/main/java/org/eclipse/lyo/server/oauth/core/utils/AbstractAdapterCredentialsFilter.java#LL358C9-L358C9

**Review by berezovskyi — CHANGES_REQUESTED** [review]

File: server/oauth-core/src/main/java/org/eclipse/lyo/server/oauth/core/utils/AbstractAdapterCredentialsFilter.java [hunk 2]
```diff
@@ -362,7 +362,7 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
 					{
 						sendUnauthorizedResponse(response, e);
 						//TODO: Change to a log message.
-			            log.info("UnauthorizedException occured while checking for Basic authentication: {}", e.getMessage());
+			            log.debug("UnauthorizedException occured while checking for Basic authentication: {}");
```
Comment (berezovskyi) [inline server/oauth-core/src/main/java/org/eclipse/lyo/server/oauth/core/utils/AbstractAdapterCredentialsFilter.java:365]: `{}` now points to a parameter that is never passed.

**Review by berezovskyi — CHANGES_REQUESTED** [review]

**Comment by berezovskyi** [comment]
> Is this really a breaking change?

I concede, it looks like a potentially breaking bugfix given the javadoc on some of the updated methods was stating that the values are stored in the session store, while they were not.

We also looked with Jad at the [potential uses](https://sourcegraph.com/search?q=context%3Aglobal+AbstractAdapterCredentialsFilter&patternType=standard&sm=1&groupBy=repo) using sourcegraph and it looks the Jama adaptor made use of it but they copypasted the code entirely, so our changes won't affect them.

**Comment by Jad-el-khoury** [comment]
Naice. It says "1 review requesting changes" but I can't seem to find any outstanding requests.

Happy to approve & merge @berezovskyi ! I canot do that myself yet.

**Pull request merged**